src/components/InvertedFlatList/BaseInvertedFlatList.js

```diff
@@ -134,6 +156,8 @@ class BaseInvertedFlatList extends Component {
             );
         }
 
+        // For native devices since we're not measuring let's make this callback sooner.
+        this.props.onMeasurementEnd(false);
```

Comment:
So from your suggestions 1 is already done as part of [these](https://github.com/Expensify/App/blob/756a9f7f1b17f87112d6d14ec49e60269a499df8/src/components/InvertedFlatList/BaseInvertedFlatList.js#L151-L153) wherein measureItemLayout will call onMeasurementEnd once all items are measured.

For your suggestion 2, `onInitialItemRendered` hmm i'm confused. Is this something you want as a callback all the way to `ReportActionsView` and then in ReportActionsView we'll set isDoneMeasuring in `onInitialItemRendered` as true there? That seems a little confusing, vs the code [here](https://github.com/Expensify/App/blob/0fb83f63bcf461d1d7777264a47ec61199355a81/src/components/InvertedFlatList/BaseInvertedFlatList.js#L151-L160) which is if shouldMeasureItems is false we call onMeasurementEnd. I could add a comment why we pass false since that is unclear. Let me know.

Reply:
>Is this something you want as a callback all the way to ReportActionsView and then in ReportActionsView we'll set isDoneMeasuring in onInitialItemRendered as true t[here](https://github.com/Expensify/App/blob/0fb83f63bcf461d1d7777264a47ec61199355a81/src/components/InvertedFlatList/BaseInvertedFlatList.js#L151-L160)? That seems a little confusing, vs the code here which is if shouldMeasureItems is false we call onMeasurementEnd.

Yep, you got it - just asking for a more descriptive callback name.

>That seems a little confusing, vs the code [here](https://github.com/Expensify/App/blob/0fb83f63bcf461d1d7777264a47ec61199355a81/src/components/InvertedFlatList/BaseInvertedFlatList.js#L151-L160) which is if shouldMeasureItems is false we call onMeasurementEnd.

We can always ask for a tie breaker on this one 😄

We have a method called `onMeasurementEnd()` which is called when **we do not measure anything at all** with an optional callback param to **let the caller know that we did not actually measure anything**. The callback name itself implies that some measuring is happening. This is less confusing than two callbacks that are named for exactly what they do?

>I could add a comment why we pass false since that is unclear. Let me know if i'm misunderstand something here.

Sure. But why not just write code that documents itself vs. document why some code says it will do one thing, but actually does something else?

Reply:
Okay cool i think i get your perspective now. Before i was doing something were i was tracking measurement and render for both web and mobile before scrolling. Now i just updated the code to remove onMeasurementEnd(boolean) for mobile and removed render tracking for web. I use `shouldTrackItemRendered` for native and `onMeasurementEnd` for web.

So now for web, once isMeasurement is done we scroll, no polling. If item is not found we load more.
For native, each render has a callback to capture renderedItems, there is polling to determine if item is rendered and we can scroll. I can update the code to fire the callback only when the desired item is rendered but then i wasn't able to find a good solution to determine when all items are rendered so that we can load more chats if needed. 
I've tried having a callback in `ReportActionsList` in componentDidMount or DidUpdate to capture when all actions were rendered but it doesn't seem to work well. didMount/didUpdate gets fired too early i.e., before the item is rendered.

Update the code. Let me know if the change looks alright.